src/mlpack/methods/kde/kde_model_impl.hpp

```diff
@@ -0,0 +1,195 @@
+/**
+ * @file kde_model_impl.hpp
+ * @author Roberto Hueso
+ *
+ * Implementation of KDE Model.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_KDE_MODEL_IMPL_HPP
+#define MLPACK_METHODS_KDE_MODEL_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "kde_model.hpp"
+
+#include <boost/serialization/variant.hpp>
+
+namespace mlpack {
+namespace kde {
+
+//! Initialize the KDEModel with the given parameters.
+inline KDEModel::KDEModel(const double bandwidth,
+                          const double relError,
+                          const double absError,
+                          const bool breadthFirst,
+                          const KernelTypes kernelType,
+                          const TreeTypes treeType) :
+  bandwidth(bandwidth),
+  relError(relError),
+  absError(absError),
+  breadthFirst(breadthFirst),
+  kernelType(kernelType),
+  treeType(treeType)
+{
+  // Nothing to do
+}
+
+// Copy constructor.
+inline KDEModel::KDEModel(const KDEModel& other) :
+  bandwidth(other.bandwidth),
+  relError(other.relError),
+  absError(other.absError),
+  breadthFirst(other.breadthFirst),
+  kernelType(other.kernelType),
+  treeType(other.treeType)
+{
+  // Nothing to do
+}
+
+// Move constructor.
+inline KDEModel::KDEModel(KDEModel&& other) :
+  bandwidth(other.bandwidth),
+  relError(other.relError),
+  absError(other.absError),
+  breadthFirst(other.breadthFirst),
+  kernelType(other.kernelType),
+  treeType(other.treeType),
+  kdeModel(std::move(other.kdeModel))
+{
+  // Reset other model
+  other.bandwidth = 1.0;
+  other.relError = 1e-6;
+  other.absError = 0;
+  other.breadthFirst = false;
+  other.kernelType = KernelTypes::GAUSSIAN_KERNEL;
+  other.treeType = TreeTypes::KD_TREE;
+  other.kdeModel = decltype(other.kdeModel)();
+}
+
+inline KDEModel& KDEModel::operator=(KDEModel other)
```

Comment:
Just to check, I think this covers both the move and copy `operator=()`.  If so that is nice and probably a simplification we could (and should) make in other places too. :+1: